File: 02_parallel/problems/00_hello_threads/hello_threads.f90
program hello_threads
  use omp_lib
  ! TODO: 下の print を !$omp parallel ... !$omp end parallel で囲み, 複数のスレッドで実行させよ.
  print "(a,i0,a,i0)", "hello from thread ", omp_get_thread_num(), &
       " of ", omp_get_num_threads()
  ! TODO: 上で始めた parallel 領域を閉じる (!$omp end parallel).
end program hello_threads

File: 02_parallel/problems/00_hello_threads/hello_threads.cpp
#include <cstdio>
#include <omp.h>

int main() {
  // TODO: 下のブロックの直前に #pragma omp parallel を1行追加し, printf を複数のスレッドで実行させよ.
  {
    printf("hello from thread %d of %d\n",
           omp_get_thread_num(), omp_get_num_threads());
  }
  return 0;
}
